bindings_node/src/mls_client.rs

```diff
@@ -0,0 +1,189 @@
+  }
+
+  #[napi]
+  pub async fn can_message(&self, account_addresses: Vec<String>) -> Result<HashMap<String, bool>> {
+    let results: HashMap<String, bool> = self
+      .inner_client
+      .can_message(account_addresses)
+      .await
+      .map_err(|e| Error::from_reason(format!("{}", e)))?;
+
+    Ok(results)
+  }
+
+  #[napi]
+  pub async fn register_ecdsa_identity(&self, signature_bytes: Uint8Array) -> Result<()> {
+    if self.is_registered() {
+      return Err(Error::from_reason(
+        "An identity is already registered with this client",
+      ));
+    }
+
+    let mut signature_request = match self.inner_client.identity().signature_request() {
+      Some(signature_req) => signature_req,
+      None => return Err(Error::from_reason("No signature request found")),
+    };
+    let signature_text = match self.signature_text() {
+      Some(text) => text,
+      None => return Err(Error::from_reason("No signature text found")),
+    };
+
+    let signature = Box::new(RecoverableEcdsaSignature::new(
+      signature_text,
+      signature_bytes.deref().to_vec(),
+    ));
+
+    signature_request
+      .add_signature(signature)
+      .await
+      .map_err(|e| Error::from_reason(format!("{}", e)))?;
+
+    self
+      .inner_client
+      .register_identity(signature_request)
+      .await
+      .map_err(|e| Error::from_reason(format!("{}", e)))?;
+
+    return Ok(());
+  }
+
+  #[napi]
+  pub async fn register_erc1271_identity(
+    &self,
+    signature_bytes: Uint8Array,
+    chain_rpc_url: String,
+  ) -> Result<()> {
+    if self.is_registered() {
+      return Err(Error::from_reason(
+        "An identity is already registered with this client",
+      ));
+    }
+
+    let mut signature_request = match self.inner_client.identity().signature_request() {
+      Some(signature_req) => signature_req,
+      None => return Err(Error::from_reason("No signature request found")),
+    };
+    let signature_text = match self.signature_text() {
+      Some(text) => text,
+      None => return Err(Error::from_reason("No signature text found")),
+    };
+
+    let signature = Box::new(
+      Erc1271Signature::new_with_rpc(
+        signature_text,
+        signature_bytes.deref().to_vec(),
+        self.account_address.clone(),
+        chain_rpc_url,
+      )
+      .await
+      .map_err(|e| Error::from_reason(format!("{}", e)))?,
+    );
+
+    signature_request
+      .add_signature(signature)
+      .await
+      .map_err(|e| Error::from_reason(format!("{}", e)))?;
+
+    self
+      .inner_client
+      .register_identity(signature_request)
+      .await
+      .map_err(|e| Error::from_reason(format!("{}", e)))?;
+
+    return Ok(());
+  }
+
+  #[napi]
+  pub fn signature_text(&self) -> Option<String> {
+    match self.inner_client.identity().signature_request() {
+      Some(signature_req) => Some(signature_req.signature_text()),
+      None => None,
+    }
+  }
```

Comment:
There might be multiple signature required from that `signature_request` hence you can't have `register_erc1271_identity` and `register_ecdsa_identity `. 
Here is a flow that makes sense to me 
```
let signature_request = client.signature_request()
let signature_text = signature_request.signature_text()
signature_request.add_signature(Box::new(Erc1271 or ECDSA signature))
client.register_identity(signature_request)
```
So you need to have a napi client that exposes `register_identity()`, `signature_request()`, and a napi siganture_request that exposes `signature_text()`, `add_signature()`m which is identical to ffi bindings.

Reply:
i originally went down this path, but passing `SignatureRequest` between node and rust creates a lot of complexity. my understanding is that only 1 signature is required per client.

happy to sync on this to better understand how this should work and how best to pass these more complex data types between node and rust.

Reply:
> my understanding is that only 1 signature is required per client

There are cases where more than 1 signature is required e.g. it requires two signatures to revoke a wallet, one from recovery address & one from revoked wallet. 
This is fine for creating new inboxes, but I'm afraid we will need to figure out complex data types very soon.